src/Nethermind/Nethermind.Synchronization/Synchronizer.cs

```diff
@@ -421,27 +225,27 @@ public Task StopAsync()
             return Task.WhenAny(
                 Task.Delay(FeedsTerminationTimeout),
                 Task.WhenAll(
-                    _fastSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    _stateSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    SnapSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    _fullSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    HeadersSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    BodiesSyncFeed?.FeedTask ?? Task.CompletedTask,
-                    ReceiptsSyncFeed?.FeedTask ?? Task.CompletedTask));
+                    fullSyncComponent.Feed.FeedTask,
+                    fastSyncComponent.Feed.FeedTask,
+                    stateSyncComponent.Feed.FeedTask,
+                    snapSyncComponent.Feed.FeedTask,
+                    fastHeaderComponent.Feed.FeedTask,
+                    oldBodiesComponent.Feed.FeedTask,
+                    oldReceiptsComponent.Feed.FeedTask));
         }
 
         private void WireMultiSyncModeSelector()
         {
-            WireFeedWithModeSelector(_fastSyncFeed);
-            WireFeedWithModeSelector(_stateSyncFeed);
-            WireFeedWithModeSelector(SnapSyncFeed);
-            WireFeedWithModeSelector(_fullSyncFeed);
-            WireFeedWithModeSelector(HeadersSyncFeed);
-            WireFeedWithModeSelector(BodiesSyncFeed);
-            WireFeedWithModeSelector(ReceiptsSyncFeed);
+            WireFeedWithModeSelector(fullSyncComponent.Feed);
```

Comment:
Can the feeds just receive selector in ctor and subscribe?

Reply:
Ideally, yea... But I'm not trying to change its behaviour here.